### PR TITLE
Bootstraped the DAV Client

### PR DESCRIPTION
**Pull request by huy-ta**
Resolves https://github.com/OpenPaaS-Suite/esn-frontend-calendar/issues/472.

- [x] Write tests (Jest)

**=========== README.md ===========**

# DAV Client

A DAV client to send requests to DAV servers.

## Usage

Here's an example usage:

```javascript
import { DAVClient, getInbox, HTTPFetchClient } from 'dav-client';

const [username, password] = ['admin@open-paas.org', 'secret'];
const httpClient = new HTTPFetchClient();
const davClient = new DAVClient({ baseURL: 'http://0.0.0.0:8001', httpClient, headers: { Authorization: `Basic ${Buffer.from([username, password].join(':')).toString('base64')}`} });

getInbox(davClient)('5f60334e78d1b021351f9f6e')
  .then(inbox => {
    console.log('inbox =', inbox);
  });
```

You can replace the default `HTTPFetchClient` (which uses [the Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API)) with any http client as long as it follows the following interface:

```typescript
export interface RequestOptions {
  url: string;
  method: string;
  params?: Record<string, unknown>;
  headers?: HeadersInit;
  body?: BodyInit | null;
}

export interface HTTPClient {
  request(options: RequestOptions): Promise<Response>;
  requestJson<T>(options: RequestOptions): Promise<T>;
  requestText(options: RequestOptions): Promise<string>;
}
```

## Build

To build the library, run the following command:

```bash
npm run build
```

The built code will reside in the **./build/** folder.

## Test

To run tests, run the following command:

```bash
npm run test
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+node_modules/
+build/
+dist/
+.env
+package-lock.json
```
Comment (rezk2ll) [inline .gitignore:5]:
```suggestion
package-lock.json
.vscode
```

File: rollup.config.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import commonjs from '@rollup/plugin-commonjs';
+import resolve from '@rollup/plugin-node-resolve';
+import { terser } from 'rollup-plugin-terser';
+import typescript from '@rollup/plugin-typescript';
+
+const production = !process.env.ROLLUP_WATCH;
+
+export default {
+	input: 'lib/index.ts',
```
Comment (rezk2ll) [inline rollup.config.js:9]: seems like 8 space tabs here, maybe we should use 2?

File: package.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "build": "rollup -c",
+    "dev": "rollup -c -w"
+  },
+  "devDependencies": {
+    "@rollup/plugin-commonjs": "^17.1.0",
+    "@rollup/plugin-node-resolve": "^11.2.0",
+    "@rollup/plugin-typescript": "^8.2.0",
+    "@tsconfig/recommended": "^1.0.1",
+    "@types/url-join": "^4.0.0",
+    "rollup": "^2.42.0",
```
Comment (renaudboyer) [inline package.json:16]: We'll use webpack as it's the standard here

File: package.json [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    "rollup-plugin-terser": "^7.0.2",
+    "tslib": "^2.1.0",
+    "typescript": "^4.2.3"
+  },
```
Comment (renaudboyer) [inline package.json:20]: We need some testing library too :wink: (jest)

File: rollup.config.js (hunk 2 above)
Comment (huy-ta) [reply rollup.config.js]: It was actually an indent using a single tab. I've added eslint and fixed the problem.

File: .gitignore (hunk 1 above)
Comment (huy-ta) [reply .gitignore:6]: Added.

File: package.json (hunk 4 above)
Comment (huy-ta) [reply package.json:23]: I'll add Jest when I start writing tests for these pieces of code (which is after you guys have finished reviewing the code :smile:).

File: package.json (hunk 3 above)
Comment (huy-ta) [reply package.json:19]: Unfortunately, Webpack doesn't support output-ing ES Modules (which is crucial for tree shaking and code splitting later on when we build our app): https://github.com/webpack/webpack/issues/2933. So we still need to use Rollup here.

**Review by rezk2ll — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (rezk2ll) [reply package.json:20]: vite is also in the table too https://vitejs.dev/

Comment (huy-ta) [reply package.json:20]: Vite is mostly for applications. It also uses Rollup to build under the hood.

File: package.json (hunk 4 above)
Comment (huy-ta) [reply package.json:29]: Jest & tests are added now.

**Pull request merged**